test/suite/test_cc02.py

```diff
@@ -114,6 +114,7 @@ def test_cc(self):
             ',stable_timestamp=' + self.timestamp_str(200))
 
         # Trigger checkpoint cleanup and wait until it is done. This should clean the history store.
+        # TODO - Why is it failing?
```

Comment:
@kommiharibabu, this passes if I trigger a checkpoint before. This means we need 2 checkpoints before being able to prune the obsolete data, is it expected?

Reply:
Sorry, the above is wrong. The test still does not pass.

Reply:
Hmm without modifying the test, sometimes it passes, sometimes it does not.

Reply:
I believe this is because we evict before obsolete cleanup processes them.

Reply:
Yes forced eviction was at play, see f368b8d4274f295cc72239cdbb697c38839db548.

Reply:
ok thanks for the investigation.